packages/cdc-react/src/components/ProfileHeader/ProfileHeaderUserProfileMenuItem/ProfileHeaderUserProfileMenuItem.tsx

```diff
@@ -0,0 +1,166 @@
+import "./ProfileHeaderUserProfileMenuItem.scss";
+
+import { PopupMenuItemType } from "../../../@types";
+
+import { useState, useEffect, useLayoutEffect, useRef } from "react";
+
+import { ProfileHeaderPopupMenuItem } from "../ProfileHeaderPopupMenuItem/ProfileHeaderPopupMenuItem";
+import { Icon } from "../../Icon/Icon";
+
+export interface ProfileHeaderUserProfileMenuItemProps {
+  className?: string;
+  avatar?: string | JSX.Element;
+  userProfilePopupMenuItems: PopupMenuItemType[];
+}
+
+interface PopupKeyCodeObject {
+  [key: string]: (
+    currentButton?: HTMLButtonElement | undefined,
+    isLastIndex?: boolean | undefined
+  ) => void;
+}
+
+/**
+ * Profile Header User Profile Menu Item UI component
+ */
+export const ProfileHeaderUserProfileMenuItem = ({
+  className, // avatar,
+  userProfilePopupMenuItems = [],
+}: ProfileHeaderUserProfileMenuItemProps) => {
+  const [popupOpen, setProfileHeaderPopupOpenState] = useState(false);
+  const [activePopupItemIndex, setProfileHeaderPopupItemIndex] = useState(0);
+
+  const popupRef = useRef<HTMLDivElement>(null);
+  const userProfileMenuItemRef = useRef<HTMLButtonElement>(null);
+
+  const handleClickOutsidePopup = (ev: React.MouseEvent) => {
+    if (
+      userProfileMenuItemRef.current &&
+      userProfileMenuItemRef.current?.contains(ev.target as Node)
+    ) {
+      setProfileHeaderPopupOpenState(!popupOpen);
+
+      // Utilizes a setTimeout to move event after react hook event
+      setTimeout(() => {
+        (
+          document.getElementsByClassName("profile-header-popup-menu-item")[
+            activePopupItemIndex
+          ] as HTMLButtonElement
+        ).focus();
+      }, 0);
+    } else if (
+      popupRef.current &&
+      !popupRef.current?.contains(ev.target as Node)
+    ) {
+      setProfileHeaderPopupOpenState(false);
+    }
+  };
+
+  const PopupKeyDownMethods: PopupKeyCodeObject = {
+    Tab: (currentButton?: HTMLButtonElement, isLastIndex?: boolean) => {
+      if (!isLastIndex) {
```

Comment:
Prefer positive logic and have `if (isLastIndex)`.  Also, the `else if` is unnecessary.  Can just be `else`